### PR TITLE
Feat: Sort search

### PR DESCRIPTION
**Pull request by bharati-21**
This PR contains the following-
- Users can sort their feed by: latest (default), oldest or trending posts.
- Users can search for any user by their first, username or last name. Debounced the search results.

### CODE REVIEW

File: src/hooks/useDebounce.jsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import React, { useCallback, useState } from "react";
+import { getUsersState } from "features";
+import { useSelector } from "react-redux";
+import { getSearchedUsers } from "utils";
+
+const useDebounce = () => {
```
Comment (ivinayakg) [inline src/hooks/useDebounce.jsx:6]: amazing this is the mind blowing

**Review by ivinayakg — APPROVED** [review]
amazing work 👍

**Review by suryxks — APPROVED** [review]
Code is readable and well structured 
Ui looks great  in dark mode
few suggestions :
1)nav bar should be sticky 
2)you can use a darker shade of the cta color  in light mode the current cta color is very light in  light mode

**Review by junaidshaikh-js — APPROVED** [review]
Great work done! 

Clean code and good profile structure. Can be merged.

File: src/hooks/useDebounce.jsx (hunk 1 above)
Comment (bharati-21) [reply src/hooks/useDebounce.jsx:6]: Thank you.

**Comment by bharati-21** [comment]
> Code is readable and well structured Ui looks great in dark mode few suggestions : 1)nav bar should be sticky 2)you can use a darker shade of the cta color in light mode the current cta color is very light in light mode

Thank you
Yes, I have made the navbar sticky.
Oh okay. I'll try to change the cta color if possible in the light mode before submission. Else, I shall do that in temp dev,

**Comment by bharati-21** [comment]
> Great work done!
> 
> Clean code and good profile structure. Can be merged.

Thank you.

**Comment by bharati-21** [comment]
> amazing work 👍

Thank you for your review!

**Pull request merged**